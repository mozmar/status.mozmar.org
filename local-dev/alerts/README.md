Alerts in this directory will not be put in source control.

Feel free to copy the alerts from `/alerts` here for testing. Or manually create
a few. Do what you feel. It's your (local) world.
