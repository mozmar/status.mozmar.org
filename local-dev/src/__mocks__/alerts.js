const alerts = [
    {
        alertLogged: '2016-12-05-17-23-45-42',
        alertOccurred: '2016-12-05-17-20-12-34',
        description: 'The decline in spirograph and the rise in gang activity',
        graph: 'https://frinkiac.com/img/S07E02/1108189.jpg',
        link: 'https://frinkiac.com/episode/S07E02/1108189',
        resolved: false,
        severity: 'h'
    },
    {
        alertLogged: '2016-12-17-16-53-12-23',
        alertOccurred: '2016-12-17-12-30-12-34',
        description: 'Disco record sales were up 400%',
        graph: 'https://frinkiac.com/img/S08E11/288721.jpg',
        link: 'https://frinkiac.com/episode/S08E11/290589',
        resolved: true,
        severity: 'l'
    },
    {
        alertLogged: '2016-12-14-12-45-12-34',
        alertOccurred: '2016-12-14-12-30-12-34',
        description: 'As intelligence goes up, happiness often goes down',
        graph: 'https://frinkiac.com/img/S12E09/1066440.jpg',
        link: 'https://frinkiac.com/episode/S12E09/1066440',
        resolved: false,
        severity: 'm'
    }
];

export default alerts;
