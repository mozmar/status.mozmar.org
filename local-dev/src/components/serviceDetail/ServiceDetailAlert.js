import React, { Component } from 'react';

import ServiceDetailGraph from './ServiceDetailGraph';

import { formatYamlDateString } from '../../helpers';

class ServiceDetailAlert extends Component {
    static propTypes = {
        alertData: React.PropTypes.object.isRequired
    }

    renderResolved(resolved) {
        // These map to Bootstrap label class names.
        const labelClass = resolved ? 'success' : 'danger';

        const label = resolved ? 'yes' : 'no';

        return (
            <span className={'label label-' + labelClass}>{label}</span>
        );
    }

    renderSeverity(severity) {
        let labelClass;
        let label;

        // These also map to Bootstrap label class names.
        switch (severity) {
            case 'h':
                labelClass = 'warning';
                label = 'high';
                break;
            case 'm':
                labelClass = 'info';
                label = 'medium';
                break;
            default:
                labelClass = 'default';
                label = 'low';
                break;
        }

        return (
            <span className={'label label-' + labelClass}>{label}</span>
        );
    }

    render() {
        const data = this.props.alertData; // because convenience

        return (
            <div className="service-detail-alert">
                <h4><a href={data.link} rel="external">{data.description}</a></h4>
                <div className="service-detail-alert-data-wrapper">
                    <dl className="service-detail-alert-data-list">
                        <dt>Resolved</dt>
                        <dd>{this.renderResolved(data.resolved)}</dd>
                        <dt>Severity</dt>
                        <dd>{this.renderSeverity(data.severity)}</dd>
                        <dt>Occurred</dt>
                        <dd>{formatYamlDateString(data.alertOccurred).toString()}</dd>
                        <dt>Logged</dt>
                        <dd>{formatYamlDateString(data.alertLogged).toString()}</dd>
                    </dl>
                    <ServiceDetailGraph url={data.graph} />
                </div>
            </div>
        );
    }
}

export default ServiceDetailAlert;
