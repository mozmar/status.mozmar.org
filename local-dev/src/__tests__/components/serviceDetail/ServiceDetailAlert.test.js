import React from 'react';
import renderer from 'react-test-renderer';

import ServiceDetailAlert from '../../../components/serviceDetail/ServiceDetailAlert';

import alerts from '../../../__mocks__/alerts';

test('render ServiceDetailAlert', () => {
    const component = renderer.create(
        <ServiceDetailAlert alertData={alerts[0]} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
