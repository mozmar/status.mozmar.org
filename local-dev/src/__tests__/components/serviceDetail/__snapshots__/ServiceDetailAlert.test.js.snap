exports[`test render ServiceDetailAlert 1`] = `
<div
  className="service-detail-alert">
  <h4>
    <a
      href="https://frinkiac.com/episode/S07E02/1108189"
      rel="external">
      The decline in spirograph and the rise in gang activity
    </a>
  </h4>
  <div
    className="service-detail-alert-data-wrapper">
    <dl
      className="service-detail-alert-data-list">
      <dt>
        Resolved
      </dt>
      <dd>
        <span
          className="label label-danger">
          no
        </span>
      </dd>
      <dt>
        Severity
      </dt>
      <dd>
        <span
          className="label label-warning">
          high
        </span>
      </dd>
      <dt>
        Occurred
      </dt>
      <dd>
        Thu Jan 05 2017 11:20:12 GMT-0600 (CST)
      </dd>
      <dt>
        Logged
      </dt>
      <dd>
        Thu Jan 05 2017 11:23:45 GMT-0600 (CST)
      </dd>
    </dl>
    <div
      className="service-detail-alert-graph-wrapper">
      <iframe
        className="service-detail-alert-graph"
        src="https://frinkiac.com/img/S07E02/1108189.jpg" />
    </div>
  </div>
</div>
`;
